### DISCUSSION TITLE
Some issues with current master

### DISCUSSION THREAD

**Discussion by spamatica** [muse-sequencer/muse#1187] [Team Posts]
I decided to make a stealthy report of some issues before I make tickets out of them.
I think all of these are old issues that I tripped on, not related to kybos UI changes.

First a question to Tim. There was an issue not that long ago about Overdub/Replace not working. I think Tim mentioned taking a look at this, did anything come out of that? I didn't test it yet, just curious.

Some other issues:
- First we have AUXes. If I'm not mistaken they are tapped before the plugins. Did this always use to be like this? I'm not sure this is a bug but I didn't find the logic in sending the audio before the plugin rack. E.g If I would add reverb with an aux but the reverb should be applied to the EQ:ed audio, in other words after the plugin rack.

- Ctrl+drag of cloned part shows a new clone in the arranger but after saving and reloading the song it is no longer a clone, something not copied correctly.. Ctrl+B works fine.

- Insert bar removes automation. Think I forgot this when I updated the structure functions.

There were a few more issues on my list but for now I don't think MusE is at fault for those.

**Comment by terminator356** [comment]
Overdub/Replace not working:
Yes I have made much progress on this and have a branch to finish up.
It only fixes the midi overdubbing for now, since that seems to be the most popular use.
It was quite an involved fix, and I need to clear up some drawing issues left.
But it was side-tracked by the recent bug reports.
I should soon be finished the most recent crash bug (when dragging a part with multiple controller
  values at the same time). I finished one part last night. Hang in there...

AUXes: No, I have never changed any fundamental behaviour in there.
I could take a look at the process engine, perhaps a simple move of the aux mix code,
 past the plugin mix code?
Please **note**: I discovered a fundamental problem with the way our AUXes work.
There is a problem with how the aux 'indexes' work. I forget the details but it is something like
 it does not correctly remember which aux was assigned to which index.
If you have two auxes, then delete the first one, the second does not properly take over that first index.
The two indexes become mixed up. The solution will be somewhat detailed, so I didn't have time to fix it.
It is not just a bug but a fundamental flaw in the way we index them.

The last two complaints not sure, don't recall breaking anything, but hey, I probably did.
Will check...

**Comment by terminator356** [comment]
AUXes flaw:
Steps to reproduce: 1 Find a partner. 2 Just do it.
But seriously. You can read all about it in my new book "How to build a staircase in 5 easy steps".
But seriously.
1) Add a Wave track and two Aux tracks Aux 1 and Aux 2 (in that order).
2) On the Wave track, adjust each Aux value differently, to help identify them and observe the flaw.
Say Aux 1 = 3 and Aux 2 = 7.
3) Delete the Aux 1 track.
4) Observe **flaw**: On the Wave track there is now (correctly) just one Aux but it is (incorrectly)
 labelled Aux 2 and its value is 3 !
5) Hit Undo.
6) Observe **flaw**: On the Wave track there are now (correctly) two Aux but they are (incorrectly)
 labelled Aux 2 and Aux 1 and their values are 3 and 7 respectively.

(This should be entered as a design flaw issue so someone can tackle it.)
Cheers.
